go/private/toolchain.bzl

```diff
@@ -12,24 +12,67 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-def _go_sdk_repository_impl(ctx):
-  goroot = ctx.path(".")
-  ctx.template("BUILD.bazel", 
-    Label("@io_bazel_rules_go//go/private:BUILD.sdk.bazel"),
-    substitutions = {"{goroot}": str(goroot)}, 
-    executable = False,
-  )
-  ctx.download_and_extract(
-      url = ctx.attr.url,
-      stripPrefix = ctx.attr.strip_prefix,
-      sha256 = ctx.attr.sha256)
+def _go_host_sdk_impl(ctx):
+  root = _detect_host_sdk(ctx)
+  _local_sdk(ctx, root)
+  _skd_build_file(ctx, root)
+
+go_host_sdk = repository_rule(_go_host_sdk_impl, environ = ["GOROOT"])
+"""go_host_sdk is a specialization of go_sdk just to add the GOROOT dependancy"""
 
+def _go_sdk_impl(ctx):
+  _skd_build_file(ctx, str(ctx.path(".")))
```

Comment:
Should this be inside the branch?

Also, will this end up being an absolute path? Not sure whether it will work with remote execution.

Reply:
Yes, it was safe but wrong.
Remote execution is a problem for another day (this does not add any new absolute paths).
I think it will be okay because under remote execution conditions you will have constructed an sdk that reports correct absolute paths for the execution container, not the local one.